Tools/Scripts/webkitpy/browserperfdash/browserperfdash_runner.py

```diff
@@ -342,11 +348,88 @@ def _run_benchmark_runner_plan(self, plan, result_file_path):
+        subtests_to_run = BenchmarkRunner.format_subtests(BenchmarkRunner.available_subtests(plan))
+        total_subtests = len(subtests_to_run)
+        _log.info(f'Running the {total_subtests} subtests of benchmark plan {plan} one-by-one')
+        current_subtest_index = 0
+        main_benchmark_name = None
+        metrics_aggregation_entry = None
+        split_subtest_data = {}
+        split_subtest_data_is_empty = True
+        subtests_passed = []
+        subtests_failed = []
+        for subtest in subtests_to_run:
+            current_subtest_index += 1
+            _log.info(f'Running subtest {subtest} of benchmark plan {plan} [{current_subtest_index} of {total_subtests}]')
+            benchmark_runner_class = benchmark_runner_subclasses[self._args.driver]
+            with tempfile.NamedTemporaryFile() as subtest_temp_result_file:
+                try:
+                    runner = benchmark_runner_class(plan,
+                                                    self._args.localCopy,
+                                                    self._args.countOverride,
+                                                    self._args.timeoutOverride,
+                                                    self._args.buildDir,
+                                                    subtest_temp_result_file.name,
+                                                    self._args.platform,
+                                                    self._args.browser,
+                                                    None,
+                                                    subtests=[subtest],
+                                                    browser_args=self._args.browser_args)
+                    runner.execute()
+                except Exception as e:
+                    _log.error(f'subtest {subtest} of benchmark plan {plan} failed with exception: {e}')
+                    subtests_failed.append(subtest)
+                    continue
+                subtest_temp_result_file.flush()
+                subtest_temp_result_file.seek(0)
+                temp_result_json = json.load(subtest_temp_result_file)
+                if 'debugOutput' in temp_result_json:
+                    del temp_result_json['debugOutput']
+
+                benchmark_names = list(temp_result_json.keys())
+                assert(len(benchmark_names) == 1), "There is more than one main benchmark name in the result data, this is unexpected"
+                if split_subtest_data_is_empty:
+                    split_subtest_data_is_empty = False
+                    main_benchmark_name = benchmark_names[0]
+                    metrics_aggregation_entry = temp_result_json[main_benchmark_name]['metrics']
+                    split_subtest_data[main_benchmark_name] = {}
+                    split_subtest_data[main_benchmark_name]['metrics'] = metrics_aggregation_entry
+                    split_subtest_data[main_benchmark_name]['tests'] = {}
```

Comment:
Another style thing, but perhaps we could construct the dict locally and then expose it? I.e.
``` python
split_subtest_data[main_benchmark_name] = {
    'metrics': temp_result_json[main_benchmark_name]['metrics'],
    'tests' = {},
}
```

It's fine the way it is, it's just that the repetition of the long names makes it feel a tad too verbose for me.